docs/kratos/self-service/flows/account-recovery-password-reset.mdx

```diff
@@ -0,0 +1,237 @@
++              "via": "email"
++            }
+           }
+         }
+       }
+       "additionalProperties": false
+     }
+   }
+ }
+```
+
+## Email templates
+
+<p>
+  <figure>
+    <img alt="Recovery email sent to unknown address" src={useBaseUrl("images/kratos/email-recovery-unknown.png")} />
+    <figcaption>
+      If the requested email address is an unknown recovery address, an account recovery attempt email is sent to that email
+      address.
+    </figcaption>
+  </figure>
+</p>
+
+This prevents account enumeration attacks as explained in this
+[blog post by Troy Hunt](https://www.troyhunt.com/website-enumeration-insanity-how-our-personal-data-is-leaked/).
+
+<p>
+  <figure>
+    <img alt="Recovery email sent to a known recovery address" src={useBaseUrl("images/kratos/email-recovery-known.png")} />
+    <figcaption>
+      If the requested email address is a known recovery address, a recovery code is sent to that email address.
+    </figcaption>
+  </figure>
+</p>
+
+The emails are using templates that can be customized as explained in
+[Customizing E-Mail Templates](../../emails-sms/custom-email-templates).
+
+## Updating credentials
+
+After the user completes the recovery successfully they receive a privileged session and are redirected to the settings page to
+update their credentials.
+
+Read more about the [privileged sessions](../../session-management/01_overview.mdx#privileged-sessions).
+
+## Invalidate other sessions
+
+In some scenarios it can be useful to revoke all active sessions once an account has been recovered. This forces anyone with
+access to the account to re-authenticate.
+
+```diff
+ selfservice:
+   flows:
+     recovery:
+       enabled: true
+       ui_url: http://127.0.0.1:4455/recovery
++      after:
++        hooks:
++        - hook: revoke_active_sessions
+```
+
+## One-time passwords vs. magic links
+
+The one-time password method (`code`) is preferred, as magic links have multiple drawbacks:
+
+- Some email virus scanners open links in emails to scan them. This invalidates the magic link and a user may not be able to
+  recover their account.
+- API flows initialized by apps on mobile phones or smart devices are not possible when using magic links.
+- Magic links often open in different browsers than the one used to initialize the flow, depending on the user's device settings.
+  This may result in a confusing user experience for your users.
+
+Ory Identities (Ory Kratos) uses one-time passwords by default. Magic links are available for legacy reasons and will be removed
+in the future.
+
+:::important
+
+When updating an existing project from the `link` method to use the `code` method, there might be changes in the behavior of the
+UI. Double-check the flow after updating your configuration.
+
+:::
+
+## Multifactor authentication
+
+By design, the account recovery process bypasses the primary account credential. You can choose to require multifactor
+authentication for account recovery.
```

Comment:
The wording on the console (network) is not great, I've created an issue on our internal repo to invert the checkboxes and thus make the wording a bit more clear. 

However, this warning should stay in place IMO, as the account recovery effectively allows a user to bypass their primary account credential (e.g. the password) to get a privileged session so that the user can update their password. This is not great either, and we are in discussions on how to change that. But that is a little bit down the road.

>Judging by the UI text from the console, this setting affects all self-service flows?

That toggle only applies to the settings flow.

>If so, I think that this section should be turned into a reusable snippet that will be used across all ssf docs. I can take care of that if I deciphered the situation correctly ;)

Not sure what you mean by this, as this section only applies to this specific interaction between the recovery and settings flow.

Reply:
The setting we're talking about is in Authentication -> 2FA ->  General settings and by that I understand that this is a setting that affects all attempts of users trying to adjust their settings, not only those attempts triggered in the account recovery flow. Correct?

Reply:
If that's the case then I think it'll be much clearer if we mentioned that alongside the content from `## Updating credentials`. Happy to do that for you to avoid comment ping-pong.

Reply:
Hmm, sure. I'll double check once you're done then. :)

Reply:
>The setting we're talking about is in Authentication -> 2FA -> General settings and by that I understand that this is a setting that affects all attempts of users trying to adjust their settings, not only those attempts triggered in the account recovery flow. Correct?

Yes, that would apply to all "Settings Flows", regardless of whether the recovery is involved or not. We should add a disclaimer here, you're right.